website/project/model.py

```diff
@@ -1541,6 +1577,15 @@ def delete_search_entry(self):
             logger.exception(e)
             log_exception()
 
+    def delete_registration_tree(self):
+        self.is_deleted = True
+        if not getattr(self.embargo, 'for_existing_registration', False):
+            self.registered_from = None
```

Comment:
Are you certain that there isn't any code that assumes that all registrations have a non-null `registered_from` field? https://github.com/samchrisinger/osf.io/blob/abfc1b5af2db754d79d9b4b4dd6d3472bda63185/website/project/model.py#L3194 looks like a possible candidate; there may be others. Please double check this. I know that on production, all registrations currently have a non-null `registered_from` value. 

Furthermore, it'd be good to document the cases where a registration may have a null `registered_from` field. You can do this in comments above the line where `registered_from` is defined.

Also, do we assume that this method is only called on registrations? If so, please document this in the docstring.